### PR TITLE
build and test script changes

### PR DESCRIPTION
**Pull request by prashant-shahi**
- Modified docker-compose.prod.yml to include the latest dgraph binary
- Replaced cd with pushd/popd
- simplified some of the commands
- Added comments and indentations
- Ran the script throught shellcheck.net and passed

---
This change is [Reviewable](https://reviewable.io/reviews/dgraph-io/ratel/198)

 
 
 
 
 
 

Ratel Preview: [Dgraph Preview](https://dgraph-ratel-9b03a11d7aabb73-65403.surge.sh/?local)

### CODE REVIEW

File: scripts/functions.sh [hunk 1]
```diff
@@ -2,72 +2,67 @@
...
 
 function doChecks {
-  if ! hash go 2>/dev/null; then
-    echo "Could not find golang. Please install Go env and try again.";
-    exit 1;
-  fi
+    if ! hash go 2>/dev/null; then
+		printf "Could not find golang. Please install Go env and try again.\n";
+		exit 1;
+	fi
```
Comment (paulftw) [inline scripts/functions.sh:30]: Seems like formatting is a bit off here

**Comment by MichelDiz** [comment]
Yes, it is different. I think Prashant had another goal with this PR. (maybe to fix some things? and use the Script that is configured in the TC).

I noticed yesterday (actually I already knew it, but I had not thought about the later problems. Because as the Script already existed, I assumed it was reliable) that the script, builds the master every time (It was in the Ratel's master pipeline, with no need?), this could cause problems for Ratel. And also it seems that the Tag Master is built manually by Daniel (What may not be ideal for testing). So I created this ticket https://dgraph.atlassian.net/browse/DGRAPH-1343

For some reason, even using Docker in the Ratel pipeline, there are scripts in the Ratel that builds Dgaph's Master directly. And that is a problem, cuz adds an extra no needed step - which increases time in the build and if there is a bug, it will block it. So a nightly version would be better. Even if it is a docker image or even providing a valid binary to download via curl (With TC greens). The nightly would be more predictable. Because we know that every X hour in such a time zone we have a valid Dgraph version (Green in TC) ready to be used. And we wouldn't have to ask anyone to build it manually.

In addition; For some reason, right after Prashant changed the Script for PRs (to be equal as Ratel's master pipeline). Some of my PRs went green automatically. That sounds like something was blocking somewhere. But the Script is flawed, due to the build master step. So we need to change this too.

Also, see this commit https://github.com/dgraph-io/ratel/pull/193/commits/d09a28f1c50599f5ec6496ad9dbbae0bdc7a8ba0
I had to add "docker compose down". I found that was a bug in Docker. Even using "force" flag, the issue wasn't gone. And other issues appeared, that seems to be an issue with master.

Finally

> 1 - Decide to stick with Docker but use nightly build instead of master.
OR
> 2 - Remove Docker, as I mentioned before, I still believe that's unnecessary complexity. As you can see in this Github Actions example #202 I use no docker and all works fine. The only change we need is to add a nightly build binary. And it will be 98% reliable.

I like the ideas behind the principles DRY, KISS, YAGNI - They have some different purpose but fits everywhere.

“Keep It Simple, Stupid” - Everyone should be able to read the code(and the pipeline) and understand it no matter what. Also be able to run any code, anywhere. This might help new devs who aren't used to work in the repo (Juniors will appreciate it and seniors will stay sane). And avoid having several lines in the readme to explain it.

> I even noticed that Seniors Dev who work on more complex projects (In Dgraph's Repo) couldn't handle Ratel's repo. Although the solution was simple. It didn't seem obvious to him.

“You Aren’t Gonna Need It” - if the objective is simple and the solution is too complex, no need to maintain. Let's revamp it.

“Don’t Repeat Yourself” - For example, in several stages, there are repeated steps. As for example "run yarn/npm". This could be resolved with just one shot.

Cheers.

**Comment by MichelDiz** [comment]
@prashant-shahi, for now, please revert the Docker part. The rest is useful. There are still details to discuss.

**Comment by paulftw** [comment]
@MichelDiz I agree with DRY, KISS and all other principles.
Let's stick to discussing specific problems and ways to fix them.

I'm OK with changing `:master` to `:nightly` and there's nothing KISS'er than changing one line in the compose file. It can be done as a separate PR.

yarn/npm -- yes, it can cause problems in theory. However, for that to become a problem we at the very least needed to have both installed on the same machine. We never installed `yarn` on the build machines and do builds from a clean tree. Anyway, I'm deleting the references to yarn from the build script.

I don't know who you are quoting when saying "Seniors Dev[sic] ... couldn't handle Ratel's repo..." but then it's immediately followed by "Although the solution was simple".
Perhaps a problem with a simple solution can be solved by two lines of a simple explanation, not a total overhaul of the build system.
With help of a community user I was able to reproduce the `go-bindata` error and wkill update the `INSTRUCTIONS.md` today. If you know of any other specific problems that tripped others on the team (and simple solutions that didn't occur to them) -- let's discuss what exactly happened and how we can make those specific solutions more obvious.

You've mentioned repeated steps in several stages. I actually don't know which steps are unnecessary. Feel free to point out those specific steps and/or send pull requests to remove redundancies.

File: scripts/functions.sh (hunk 1 above)
Comment (prashant-shahi) [reply scripts/functions.sh:35]: Done

**Pull request merged**